### PR TITLE
Improve editor open URL command to open the selected portion of URL

### LINKED ISSUE #21718: `editor: open url` command should only open the selected portion of a url
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

image
this for example should only open `https://google.com` but it currently uses the entire URL. vim and nvim both handle this as i would expect

### Environment

Zed: v0.166.0 (Zed Dev a30ea2fc682bdaec572f3e130631e297670612a6)
OS: Linux Wayland alpine 3.21.0
Memory: 15.6 GiB
Architecture: x86_64
GPU: AMD Radeon RX 6950 XT (RADV NAVI21) || radv || Mesa 24.2.8

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by smitbarmase**
Closes #21718

Just like in Vim, if a URL is selected, it opens exactly that portion of the URL. Otherwise, if only the cursor is on a URL, it opens the entire URL.

Zed currently does the latter. This PR also adds support for the former.

https://github.com/user-attachments/assets/8bdd2952-ceec-487c-b27a-5cea4258eb03

Release Notes:

- Updated the `editor: open url` to also handle the selected portion of a URL.

### CODE REVIEW

**Comment by 0xtimsb** [comment]
Some edge cases I tested apart from obvious ones:

- Leading/trailing whitespace around a selected URL is allowed
- Only single URL should be selected
- Both forward and backward selections should work

File: crates/editor/src/editor.rs [hunk 1]
```diff
@@ -9241,23 +9241,45 @@ impl Editor {
...
-                editor.update(&mut cx, |_, cx| {
-                    cx.open_url(&url);
-                })
-            } else {
-                Ok(())
-            }
-        })
-        .detach();
+        let url_finder = if head == tail {
+            cx.spawn(|editor, mut cx| async move {
```
Comment (mikayla-maki) [inline crates/editor/src/editor.rs:9255]: Could we reduce the duplication here by moving the `head == tail` check inside the spawn call?

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

**Comment by 0xtimsb** [comment]
Thanks! Let me know if this works.

**Review by mikayla-maki — APPROVED** [review]

**Comment by mikayla-maki** [comment]
Thanks!

**Pull request merged**